arches/app/models/resource.py

```diff
@@ -190,15 +184,36 @@ def bulk_save(resources, flat=False):
             tiles = [tile for resource in resources for tile in resource.tiles]
 
         # need to save the models first before getting the documents for index
-        Resource.objects.bulk_create(resources)
-        TileModel.objects.bulk_create(tiles)
+        # need to handle if the bulk load is appending tiles to existing resources/
+        existing_resources_ids = Resource.objects.filter(
+            resourceinstanceid__in=[resource.resourceinstanceid for resource in resources]
+        ).values_list("resourceinstanceid", flat=True)
 
-        for resource in resources:
-            resource.save_edit(edit_type="create")
+        resources_to_update = [resource for resource in resources if resource.resourceinstanceid in existing_resources_ids]
+        resources_to_create = [resource for resource in resources if resource.resourceinstanceid not in existing_resources_ids]
 
-        resources[0].tiles[0].save_edit(note=f"Bulk created: {len(tiles)} for {len(resources)} resources.", edit_type="bulk_create")
+
+        Resource.objects.bulk_create(resources_to_create)
+        TileModel.objects.bulk_create(tiles)
+        
+        for resource in resources_to_update:
+            resource.save_edit(edit_type="append", transaction_id=transaction_id)
+        
+        for resource in resources_to_create:
+            resource.save_edit(edit_type="create", transaction_id=transaction_id)
 
         for resource in resources:
+            try:
+                resource.tiles[0].save_edit(
+                    note=f"Bulk created: {len(resource.tiles)} for resourceid {resource.resourceinstanceid} ",
+                    edit_type="bulk_create",
+                    transaction_id=transaction_id,
+                )
+            except:
+                pass
+
+        # refetch new AND updated resources and index
+        for resource in Resource.objects.filter(resourceinstanceid__in=[resource.resourceinstanceid for resource in resources]):
```

Comment:
@whatisgalen I also found that indexing was problematic for updated resources. Because we are in the middle of a transaction, requesting all the resources from the DB here brings back the pre-transaction state.

I didn't get around to fixing this, I just added my own prevent_indexing clause so that all of it was indexed at the end.

I think that we'll need to take the input resources list and - for the existing resources that are being updated - merge in the tiles from the database before indexing.

Reply:
interesting, couldn't we just change the argument `fetchTiles=True` ? I don't know what the performance implications would be but it would do what you're saying it needs to do (i.e. re-query the db for all tiles belonging to an existing resource)

Reply:
like so:
`# refetch new AND updated resources and index
        fetchTiles = len(resources_to_update) > 0
        for resource in Resource.objects.filter(resourceinstanceid__in=[resource.resourceinstanceid for resource in resources]):
            document, terms = resource.get_documents_to_index(
                fetchTiles=fetchTiles, datatype_factory=datatype_factory, node_datatypes=node_datatypes
            )`

Reply:
actually I think this is slightly optimal
```
# refetch new AND updated resources and index
        for resource in Resource.objects.filter(resourceinstanceid__in=[resource.resourceinstanceid for resource in resources]):
            fetchTiles = resource.resourceinstanceid in existing_resources_ids
            document, terms = resource.get_documents_to_index(
                fetchTiles=fetchTiles, datatype_factory=datatype_factory, node_datatypes=node_datatypes
            )
            
```

Reply:
I had not spotted that fetchTiles option. If it does fetch the existing tiles in addition to the new tiles then that should work.